examples/ota-provider-app/ota-provider-common/OTAProviderExample.cpp

```diff
@@ -95,23 +97,72 @@ EmberAfStatus OTAProviderExample::HandleQueryImage(chip::app::CommandHandler * c
                                                    const chip::app::ConcreteCommandPath & commandPath,
                                                    const QueryImage::DecodableType & commandData)
 {
-    // TODO: add confiuration for returning BUSY status
-
-    OTAQueryStatus queryStatus  = OTAQueryStatus::kNotAvailable;
     uint32_t newSoftwareVersion = commandData.softwareVersion + 1; // This implementation will always indicate that an update is
                                                                    // available (if the user provides a file).
     constexpr char kExampleSoftwareString[] = "Example-Image-V0.1";
-    bool userConsentNeeded                  = false;
     uint8_t updateToken[kUpdateTokenLen]    = { 0 };
     char strBuf[kUpdateTokenStrLen]         = { 0 };
     char uriBuf[kUriMaxLen]                 = { 0 };
+    uint32_t delayedActionTimeSec           = mDelayedActionTimeSec;
+    bool requestorCanConsent                = commandData.requestorCanConsent.ValueOr(false);
+
+    OTAQueryStatus queryStatus;
+    QueryImageResponse::Type response;
+
+    switch (mQueryImageBehavior)
+    {
+    case kRespondWithUnknown:
+        queryStatus = (strlen(mOTAFilePath) != 0) ? OTAQueryStatus::kUpdateAvailable : OTAQueryStatus::kNotAvailable;
 
-    GenerateUpdateToken(updateToken, kUpdateTokenLen);
-    GetUpdateTokenString(ByteSpan(updateToken), strBuf, kUpdateTokenStrLen);
-    ChipLogDetail(SoftwareUpdate, "generated updateToken: %s", strBuf);
+        if (queryStatus == OTAQueryStatus::kUpdateAvailable && mUserConsentDelegate != nullptr)
```

Comment:
Suggest making user consent configurable from command line. It does not have to be done in this PR. Can be followed up in https://github.com/project-chip/connectedhomeip/issues/11393

Reply:
cli option for user consent is added in this PR.